src/gp_instance.cpp

```diff
@@ -174,3 +175,57 @@ void GPInstance::EstimateSBNParameters() {
   double marginal_log_lik = GetEngine()->GetLogMarginalLikelihood();
   std::cout << std::setprecision(9) << marginal_log_lik << std::endl;
 }
+
+RootedTreeCollection GPInstance::GenerateCompleteRootedTreeCollection() {
+  Tree::TreeVector tree_vector;
+  Node::NodePtrVec topologies = dag_.GenerateAllGPNodeIndexedTopologies();
+  const EigenVectorXd gpcsp_indexed_branch_lengths = engine_->GetBranchLengths();
+
+  // Map from Leaves-encoding to parent-subsplit encoding.
+  std::unordered_map<const Node *, Bitset> leaves_to_subsplit_indexer;
+  for (const auto &topology : topologies) {
+    topology->PreOrder([this, &leaves_to_subsplit_indexer](const Node *node) {
+      GPDAGNode *dag_node = dag_.GetDagNode(node->Id());
+      if (leaves_to_subsplit_indexer.count(node) == 0) {
+        SafeInsert(leaves_to_subsplit_indexer, node, dag_node->GetBitset());
```

Comment:
You describe this as a map from a leaves encoding to a parent-subsplit encoding, but actually the mapping is from an entire Node, which has other things besides the leaves encoding. This is what you wanted?

Reply:
Yes. I was thinking to use node->Leaves() initially but realized that the node ptr is also unique. We could go back to node->Leaves() as key or remove the comment "leaves encoding" to "node ptr".

Reply:
Storing a pointer as a key to a map feels strange to me. Change however you like and I'll have a look in the AM.